File: core/java/src/main/java/hara/lang/protocol/IPopFirst.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ipopfirst", name = "IPopFirst")
public interface IPopFirst {
  @HaraMethod(value = "pop-first", arity = 1)
  IPopFirst popFirst();
}

File: core/rust/src/lang/protocol/ipopfirst.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.ipopfirst", name = "IPopFirst")]
pub trait IPopFirst {
    type Output;

    #[hara_method(value = "pop-first", arity = 1)]
    fn pop_first(&self) -> Self::Output;
}
